### ISSUE TITLE
distill_adv does not work may be owning to this?

### ISSUE THREAD

**Issue by byteghy** [hao-ai-lab/FastVideo#106]
```
     with torch.no_grad():
        fake_features = teacher_transformer(
            sample_fake,
            encoder_hidden_states,
            timestep,
            encoder_attention_mask,
            output_features=True,
            output_features_stride=discriminator_head_stride,
            return_dict=False,
        )[1]
        real_features = teacher_transformer(
            sample_real,
            encoder_hidden_states,
            timestep,
            encoder_attention_mask,
            output_features=True,
            output_features_stride=discriminator_head_stride,
            return_dict=False,
        )[1]

    fake_outputs = discriminator(fake_features)
    real_outputs = discriminator(real_features)
 ```
 above is the code for discriminator training.

however, the "teacher_transformer" module is used to predict teacher_output, which is the noise level output? therefore, for the above disc training code, the discriminator is used to judge which noise output is better? maybe this job is too hard for the disc to learn?
    
can we use x_prev = solver.euler_step(noisy_model_input, teacher_output, index) as the disc input instead of directly using teacher_output? is this a better choice? since x_prev is the latent level output, and might be more reasonable?
    
am i correct? look forward to further discussion!

**Comment by byteghy** [comment]
oh, this comment maybe non-sense. since the transform is only a linear transformer....
so, why adv loss cannot work?

**Comment by rlsu9** [comment]
Thanks for your suggestions, actually we have tried it out while the output with GAN doesn't outperform the normal Consistency distillation.
You can refer the [following script ](https://github.com/hao-ai-lab/FastVideo/blob/hy_experiments2/scripts/hunyuan_experiments_14.sh)to launch the distill_adv experiment.
